appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/DesignToolbar.java

```diff
@@ -565,17 +367,13 @@ public void removeScreen(long projectId, String name) {
     project.removeScreen(name);
   }
 
-  private void toggleEditor(boolean blocks) {
-    setButtonEnabled(WIDGET_NAME_SWITCH_TO_BLOCKS_EDITOR, !blocks);
-    setButtonEnabled(WIDGET_NAME_SWITCH_TO_FORM_EDITOR, blocks);
+  public void toggleEditor(boolean blocks) {
+    setButtonEnabled(switchToBlocks.getName(), !blocks);
+    setButtonEnabled(switchToDesign.getName(), blocks);
 
-    if (AppInventorFeatures.allowMultiScreenApplications() && !isReadOnly) {
-      if (getCurrentProject() == null || "Screen1".equals(getCurrentProject().currentScreen)) {
-        setButtonEnabled(WIDGET_NAME_REMOVEFORM, false);
-      } else {
-        setButtonEnabled(WIDGET_NAME_REMOVEFORM, true);
-      }
-    }
+    boolean notOnScreen1 = getCurrentProject() != null
+        && !"Screen1".equals(getCurrentProject().currentScreen);
+    setButtonEnabled(WIDGET_NAME_REMOVEFORM, notOnScreen1);
```

Comment:
We need to make sure that if the buttons are correctly removed above when in read only mode (which they aren't) that this code doesn't throw an error.